crates/tabby-common/src/config.rs

```diff
@@ -145,6 +146,12 @@ impl Default for ServerConfig {
 #[async_trait]
 pub trait RepositoryAccess: Send + Sync {
     async fn list_repositories(&self) -> Result<Vec<RepositoryConfig>>;
+
+    fn start_snapshot(&self) -> String {
+        Default::default()
+    }
+    fn process_file(&self, _version: String, _file: SourceFile) {}
+    fn finish_snapshot(&self, _version: String) {}
```

Comment:
Please follow my suggestion. Use pass-in version number directly as u64 (might even make it explict beying something like timestamp_version).